src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala

```diff
@@ -86,6 +86,7 @@ trait Ops[
     }
   }
 
+  // Ignoring unmatched messages only works if we force close the ocpp connection.
   def expectAllIgnoringUnmatched[T](expectations: IncomingMessageProcessor[T]*)(implicit awaitTimeout: AwaitTimeout): Seq[T] = {
```

Comment:
@reinierl I fixed the do-a-transaction example to handle the case of getting a ChangeConfigurationReq message before the StartTransactionRes. I am not sure if in OCPP this is even allows, but if it is, then we might consider a fix in docile-charger. The fix I would propose would be something along the lines of: 
```
  def sendSync[REQ <: CentralSystemReq, RES <: CentralSystemRes : ClassTag](
    req: REQ
  )(
    implicit
    reqRes: CentralSystemReqRes[REQ, RES],
    awaitTimeout: AwaitTimeout
  ): RES = {
    self.send(req)
    // self.expectIncoming(matching { case res: RES => res })
    self.expectAllIgnoringUnmatched(matching { case res: Res => res }).head
  }
```

Reply:
I would go a bit further and say that the proper fix would be to check the correlation of a response with a request. And then also to skip ahead in the queue of incoming messages to find a response for a given request.

I have heard from other people too about this problem of requests to the charge point messing up the charge point's expectations of responses to its requests. I hope I'll find some time in the coming weeks to come up with a solution.